jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -56,8 +54,8 @@ object XGBoost extends Serializable {
   private val logger = LogFactory.getLog("XGBoostSpark")
 
   private def removeMissingValues(
-      denseLabeledPoints: Iterator[XGBLabeledPoint],
-      missing: Float): Iterator[XGBLabeledPoint] = {
+      denseLabeledPoints: Seq[XGBLabeledPoint],
```

Comment:
if you take sequence, that means you will load a partition into memory entirely which will leads to OOM potentially